### ISSUE TITLE
Error when trying to pull from remote private repo

### ISSUE THREAD

**Issue by h4wkmoon** [node-red/node-red#4058]
### Current Behavior

Same as #3494

I've create a private gitlab repo, self-hosted.
Push from nodered to repo are fine.
pull gives error, both in stdout and a popup in ui.
`11 Feb 17:04:39 - [error] TypeError: Cannot read properties of undefined (reading 'fetch')
`

Manual git pull from cli inside the container work.
```
export GIT_SSH_COMMAND="ssh -o IdentitiesOnly=yes -i /data/projects/.sshkeys/admin_gitlab"
cd /data/projects/BBrose/
git pull
Updating 1dbf520..46cb728
Fast-forward
 flows.json | 7 +++++++
 1 file changed, 7 insertions(+)
```
I just have to restart the flow to apply the changes

### Expected Behavior

pull successful

### Steps To Reproduce

1. Create a new project
2. Create a repo on GitHub
3. Add the repo to Git remotes
4. Add nodes to a flow
5. Commit the changes
6. Push the code to the remote repo
7. Make changes on the remote repo
8. Refresh commit history
9. Pull changes

### Environment

- Node-RED version: 3.0.2 from docker image
- Node.js version: v16.16.0
- npm version: 8.11.0 
- Platform/OS: Docker 
- Browser: Chrome / linux desktop

**Comment by dleidert** [comment]
We are affected by this too. The repository is a private one on a self-hosted gitlab instance. Due to using the dockerized version, downgrading nodejs is not an option. If I can provide any information, please let me know.

**Comment by hardillb** [comment]
@dleidert Can you confirm which version of Node-RED/Docker container you are using.

Also can you run the following commands inside the running container:

- `cd /data/projects/<projectName>`
- `git remote -v`

You can get a shell inside the running container by running `docker exec -it <name of container> /bin/bash

**Comment by dleidert** [comment]
@hardillb Sure. The container is the one currently tagged `3.0`. According to dockerhub, that should be `3.0.2-16` (Labels: build-date: 2024-01-24T11:48:25Z and version: 3.0.2).

```
git remote -v
gitlab_ours	git@<domain>:path/to/project.git (fetch)
gitlab_ours	git@<domain>:path/to/project.git (push)
```

I have anonymized the output. There is no problem running git pull via the shell. Same as the op of this issue report.

The error message is not saying much. Is there a place where I find more information, e.g. the full backtrace?

**Comment by dleidert** [comment]
If it helps, I can give you a test account to our gitlab instance.

**Comment by hardillb** [comment]
Thanks @dleidert 

Any chance you could test on a 4.0.2 or 3.1.11 container?

**Comment by dleidert** [comment]
Yes, I can do that. However, our production systems use 3.0. Thus, it would be nice if this issue could be fixed there. I'll send you feedback for 3.1 and 4.0 soon.

**Comment by hardillb** [comment]
@dleidert please be aware you should make plans to upgrade production, 3.0.x is out of support, the 3.0.x line was replaced by 3.1.x which will be supported until mid next year.

**Comment by dleidert** [comment]
Ok. I'll forward this info and check out the release information.

**Comment by supcik** [comment]
I had a similar issue and the problem was with the ssh private key file permissions.

The permissions were : `-rwxr-xr-x+ 1 1000 1000 ... name_name`

I fixed the permission with `chmod u=rw,go= name_name` and then it worked.

Now the permissions are : `-rw------- 1 1000 1000 ... name_name`

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
